File: JAVA/165. Compare Version Numbers.java
class Solution {
    public int compareVersion(String version1, String version2) {
        int i = 0;
        int j = 0;
        int n1 = version1.length();
        int n2 = version2.length();        
        while (i < n1 || j < n2) {
            StringBuilder a = new StringBuilder();
            StringBuilder b = new StringBuilder();
            while (i < n1 && version1.charAt(i) != '.') {
                a.append(version1.charAt(i));
                i++;
            }
            while (j < n2 && version2.charAt(j) != '.') {
                b.append(version2.charAt(j));
                j++;
            }
            int a1 = a.length() > 0 ? Integer.parseInt(a.toString()) : 0;
            int b1 = b.length() > 0 ? Integer.parseInt(b.toString()) : 0;
            if (a1 < b1) {
                return -1;
            } else if (a1 > b1) {
                return 1;
            }
            i++;
            j++;
        }
        return 0;
    }
}

File: C++/165. Compare Version Numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        int i=0, j=0, ans1=0, ans2=0;
        while(i<version1.size() || j<version2.size()){
            while(i<version1.size() && version1[i]!='.'){
                ans1*=10;
                ans1+=(version1[i]-'0');
                i++;
            }
            while(j<version2.size() && version2[j]!='.'){
                ans2*=10;
                ans2+=(version2[j]-'0');
                j++;
            }
            if(ans1<ans2){return -1;}
            if(ans1>ans2){return 1;}
            i++;
            j++;
            ans1=0;
            ans2=0;
        }
        return 0;
    }
};
